### DISCUSSION TITLE
winmerge shows (differences) rotated image

### DISCUSSION THREAD

**Discussion by dale810** [WinMerge/winimerge#18] [Q&A]
Some files image compare fails, one of the images displays as rotated 90 degrees. But if I look at the preference details for the two files they match and if I view them using the standard windows image viewer the images look the same not rotated. This is on current windows 10 19041.1052
WinMerge 2.16.12.0 X64
Here are two files that show the problem.
![Img0045](https://user-images.githubusercontent.com/85809530/121790206-a0ed9f80-cba2-11eb-996e-b8e89bb90942.JPG)
![Img0084](https://user-images.githubusercontent.com/85809530/121790216-b9f65080-cba2-11eb-831e-d267ee29cf88.JPG)

Any ideas?

**Comment by dale810** [comment]
Well now I feel really dumb. I was going to show the WinMerge result but the problem just vanished. I have been getting the same file differences for many runs and when I tried to run it again to get a screen capture the problem vanished - the files compare.
If I can repeat the problem again I will post update but for now I don't have proof. So just ignore this for now.

**Comment by dale810** [comment]
Well this fails...
![Img0044](https://user-images.githubusercontent.com/85809530/121790472-591c4780-cba5-11eb-80ba-5bc3ce9fcb65.JPG)
![Img0083](https://user-images.githubusercontent.com/85809530/121790475-62a5af80-cba5-11eb-9fb5-ad3674e5831f.JPG)
and the winmerge results
![Screenshot (1)](https://user-images.githubusercontent.com/85809530/121790489-79e49d00-cba5-11eb-9c24-4d706bc7f20f.png)

**Comment by dale810** [comment]
The first case does also fail my windows got mixed up when I tried to repeat the test. The winmerge result for the pair of files is here
![Screenshot (3)](https://user-images.githubusercontent.com/85809530/121790618-b1077e00-cba6-11eb-9963-104b07bc9597.png)

**Comment by sdottaka** [comment]
The original image on the left seems to be a horizontally long image. It seems that need to change the display orientation by looking at the Exif orientation metadata.

![image](https://user-images.githubusercontent.com/98126/121791465-e783de00-cc24-11eb-8ead-75e265cb2c24.png)

I will try to fix it later

**Comment by dale810** [comment]
Just looking at the file preferences info via windows file explorer the details tab shows the same height (and width) for both files, also using the standard windows photo viewer both files display with the same orientation. As far as I can tell the files have only been rotated via the Windows file explorer dropdown menu rotate left/right. I don't know anything about exif metadata or how it would have been generated. I used to use Unix and and a different set of tools but since retirement I've resigned myself to the Windows world. I don't think these image files have existed on anything besides Windows. I have tried rotating both ways using file explorer and the image orientation follows for photo viewer and the thumbnail file explorer display but never changes for winmerge which makes sense for some kind of meta data that Windows is ignoring, just don't know how it got in there or how to change it.

Sent from Yahoo Mail on Android 
 
  On Sat, Jun 12, 2021 at 19:01, Takashi ***@***.***> wrote:   


The original image on the left seems to be a horizontally long image. It seems that need to change the display orientation by looking at the Exif orientation metadata.



I will try to fix it later
